contracts/land/LANDRegistry.sol

```diff
@@ -200,18 +223,76 @@ contract LANDRegistry is Storage,
     emit UpdateOperator(assetId, operator);
   }
 
+  //
+  // Estate generation
+  //
+
+  event EstateRegistrySet(address indexed registry);
+
+  function setEstateRegistry(address registry) onlyProxyOwner external {
+    estateRegistry = IEstateRegistry(registry);
+    emit EstateRegistrySet(registry);
+  }
+
+  function createEstate(
+    int[] x,
+    int[] y,
+    address beneficiary
+  )
+    external
+    returns (uint256)
+  {
+    require(x.length == y.length);
+    require(address(estateRegistry) != 0);
+
+    uint256 estateTokenId = estateRegistry.mintNext(beneficiary);
+
+    for (uint i = 0; i < x.length; i++) {
+      uint256 tokenId = _encodeTokenId(x[i], y[i]);
+      _moveToken(
+        _ownerOf(tokenId),
+        address(estateRegistry),
+        tokenId,
+        toBytes(estateTokenId),
+        this,
+        true
+      );
+    }
+
+    return estateTokenId;
+  }
+
+  function toBytes(uint256 x) internal pure returns (bytes b) {
+    b = new bytes(32);
+    // solium-disable-next-line security/no-inline-assembly
+    assembly { mstore(add(b, 32), x) }
+  }
+
   //
   // LAND Update
   //
 
   function updateLandData(int x, int y, string data) external onlyUpdateAuthorized (_encodeTokenId(x, y)) {
     return _updateLandData(x, y, data);
   }
+
   function _updateLandData(int x, int y, string data) internal onlyUpdateAuthorized (_encodeTokenId(x, y)) {
     uint256 assetId = _encodeTokenId(x, y);
-    _update(assetId, data);
+    address owner = _holderOf[assetId];
 
-    emit Update(assetId, _holderOf[assetId], msg.sender, data);
+    if (owner == address(estateRegistry)) {
```

Comment:
Hmm Why if you want to update your land's data, you will update the estate's metadata?

I think we can do:

### EstateAssetRegistry:
```
/**
 * 
 * Update the land metadata which is part of an estate
 * 
 * @param uint256 - estateId
 * @param uint256 - tokenId (LAND)
 * @param string - metadata
 */
function updateLandData(uint256 estateId, uint256 tokenId, string metadata) external onlyUpdateAuthorized(estateId) {
  // We could move the two following lines to a function or modifier which checks if the tokenId (land)
  // is part of the estate
  mapping(uint256 => uint256) tokenIndex = estateTokenIndex[estateId]; 
  require(tokenIndex[tokenId] != 0);
  (x, y) = dar.decodeTokenId(tokenId);
  return dar.updateLandData(x, y, metadata)
}
```

### LandAssetRegistry
Leave as it was

```
function _updateLandData(int x, int y, string data) internal onlyUpdateAuthorized (_encodeTokenId(x, y)) {
    uint256 assetId = _encodeTokenId(x, y);
      _update(assetId, data);
    }

    emit Update(
      assetId,
      _holderOf[assetId],
      msg.sender,
      data
    );
  }

```